src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -189,10 +189,12 @@ static int recv_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq, int ev
         /* NOTE: use target rank, reply to src */
         int vni_src = MPIDI_OFI_get_vni_src(c, c->rank, rreq->status.MPI_TAG);
         int vni_dst = MPIDI_OFI_get_vni_dst(c, c->rank, rreq->status.MPI_TAG);
-        MPIDI_OFI_CALL_RETRY(fi_tinjectdata(MPIDI_OFI_global.ctx[vni_dst].tx, NULL /* buf */ ,
+        int vni_local = vni_dst;
+        int vni_remote = vni_src;
+        MPIDI_OFI_CALL_RETRY(fi_tinjectdata(MPIDI_OFI_global.ctx[vni_local].tx, NULL /* buf */ ,
                                             0 /* len */ ,
                                             MPIR_Comm_rank(c),
-                                            MPIDI_OFI_comm_to_phys(c, r, vni_dst, vni_src),
+                                            MPIDI_OFI_comm_to_phys(c, r, vni_local, vni_remote),
                                             ss_bits), vni_dst, tinjectdata, FALSE /* eagain */);
```

Comment:
vni_dst to vni_local here in `vni_dst, tinjectdata, FALSE /* eagain */);`

Reply:
Can't believe I missed it! Thanks.